whitelist-authenticator/src/main/java/de/intension/authentication/ConfigurableIdpHintParamIdentityProviderAuthenticator.java

```diff
@@ -0,0 +1,105 @@
+package de.intension.authentication;
+
+import java.net.URI;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import org.jboss.logging.Logger;
+import org.keycloak.OAuth2Constants;
+import org.keycloak.authentication.AuthenticationFlowContext;
+import org.keycloak.authentication.AuthenticationProcessor;
+import org.keycloak.authentication.authenticators.browser.IdentityProviderAuthenticator;
+import org.keycloak.authentication.authenticators.browser.IdentityProviderAuthenticatorFactory;
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.protocol.oidc.OIDCLoginProtocol;
+import org.keycloak.services.Urls;
+import org.keycloak.services.managers.ClientSessionCode;
+
+/**
+ * Same as {@link IdentityProviderAuthenticator} but with configurable IdP hint
+ * parameter name instead of the hardcoded 'kc_idp_hint'.
+ */
+public class ConfigurableIdpHintParamIdentityProviderAuthenticator
+        extends IdentityProviderAuthenticator
+{
+
+    private static final Logger LOG = Logger.getLogger(ConfigurableIdpHintParamIdentityProviderAuthenticator.class);
+
+    @Override
+    public void authenticate(AuthenticationFlowContext context)
+    {
+        String idpHintParamName = context.getAuthenticatorConfig().getConfig().get(WhitelistConstants.IDP_HINT_PARAM_NAME);
+        if (context.getUriInfo().getQueryParameters().containsKey(idpHintParamName)) {
```

Comment:
maybe we should check the "idpHintParamName" whether it is empty or null

Reply:
Same as above :grin: